app/src/main/java/com/internship/move/OnboardingFragment.kt

```diff
@@ -0,0 +1,33 @@
+package com.internship.move
+
+import android.os.Bundle
+import android.view.View
+import androidx.activity.addCallback
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import com.internship.move.databinding.FragmentOnboardingBinding
+import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
+
+class OnboardingFragment : Fragment(R.layout.fragment_onboarding) {
+
+    private val binding by viewBinding(FragmentOnboardingBinding::bind)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initButtons()
+        handleBackPress()
+    }
+
+    private fun handleBackPress() {
+        requireActivity().onBackPressedDispatcher.addCallback(this) {
```

Comment:
Nice try :P but I'd like you to implement this requirement using navigation component.

As a side note, using `viewLifecycleOwner` is prefered  over `this` in a fragment
https://cs-ibrahimyilmaz.medium.com/viewlifecycleowner-vs-this-a8259800367b